pcsx2/gui/SysState.cpp

```diff
@@ -257,7 +276,13 @@ class SavestateEntryPack : public ScopedAlloc<const BaseSavestateEntry*>
 		this->operator[](i++)	= new PluginSavestateEntry( PluginId_FW );
 		this->operator[](i++)	= new PluginSavestateEntry( PluginId_DEV9 );
 	}
-	
+
+	~SavestateEntryPack() {
+		for (size_t i = 0; i < this->GetLength(); i++) {
+			delete this->operator[](i);
+		}
+	}
+
```

Comment:
As an alternative solution, maybe `SavestateEntries` could be converted to an `array` of `unique_ptr<BaseSavestateEntries>`?